utils/docker/Dockerfile.centos.7

```diff
@@ -12,13 +12,44 @@ LABEL maintainer="daos@daos.groups.io"
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# Need to be able to use a local repository for packages
+ARG REPO_URL=""
+ARG REPO_EL7=""
+
+# Install EPEL and switch to dnf as it seems a bit faster.
+RUN yum clean all &&                                            \
+    yum -y install dnf epel-release yum-utils &&                \
+    yum clean all
+
+# If a local repository is supplied, then we should use it
+# and disable using the distro repositories.
+RUN if [ -n "$REPO_URL" ]; then                                 \
+      MY_REPO="${REPO_URL}${REPO_EL7}/";                        \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+exclude=fuse-3* fuse-devel-3* fuse-libs-3* fuse-debuginfo-3* ompi*\n\
+gpgcheck=0\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      yum clean all;                                            \
```

Comment:
Why are we using `yum` here if we installed and intend to use `dnf`?

Reply:
Did not take the time to test that with DNF.